app/controllers/purchases_controller.rb

```diff
@@ -1,15 +1,19 @@
 class PurchasesController < ApplicationController
+  @@purchase_limit = 20
```

Comment:
I haven't seen this way of doing stuff before, interesting. Where did you come across it?

I was wondering about a private method `purchase_limit` that takes the params and if the necessary one isn't there, defaults it to 20. Then you could do e.g. `@purchase_limit = purchase_limit(params)`

What do you think?

Reply:
Sadly I ended up closing a lot of my research windows in an effort to tidy everything up earlier - it was one of many approaches I was figuring out (alongside hidden fields, attr_accessor things), but it seemed to answer what I wanted - having a value not tied to individual Purchase ActiveRecords that was accessible from all areas of the controller and had a default value.
I'm going to push the changes I've made regarding the other comments, and then tinker with your suggestion.

Reply:
I haven't spent a lot of time, but I'm not having much luck - the index action in the controller has a "new" set of params with no limit in it, so it's always resulting in 20; without the class variable as a way to "store" a value outside of each individual controller method, I'm not sure how I can save/store/pass it around.